IllegalStateException in getSupportRequestManagerFragment

Since I've migrated more Glide calls to use fragment instead of application context I now receive a new crash report :


```
java.lang.RuntimeException: Unable to destroy activity xxx.ui.MediasListActivity}: java.lang.IllegalStateException: Activity has been destroyed
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4199)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4217)
       at android.app.ActivityThread.access$1500(ActivityThread.java:177)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1502)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:5942)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
       at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:117)
Caused by: java.lang.IllegalStateException: Activity has been destroyed
       at android.support.v4.app.FragmentManagerImpl.enqueueAction(FragmentManager.java:1399)
       at android.support.v4.app.BackStackRecord.commitInternal(BackStackRecord.java:637)
       at android.support.v4.app.BackStackRecord.commitAllowingStateLoss(BackStackRecord.java:620)
       at com.bumptech.glide.manager.RequestManagerRetriever.getSupportRequestManagerFragment(RequestManagerRetriever.java:187)
       at com.bumptech.glide.manager.SupportRequestManagerFragment.onAttach(SupportRequestManagerFragment.java:116)
       at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:907)
       at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1138)
       at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:740)
       at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1501)
       at android.support.v4.app.FragmentManagerImpl.dispatchDestroy(FragmentManager.java:1965)
       at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1088)
       at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1138)
       at android.support.v4.app.FragmentManagerImpl.moveToState$2563266(FragmentManager.java:1120)
       at android.support.v4.app.FragmentManagerImpl.dispatchDestroy(FragmentManager.java:1966)
       at android.support.v4.app.FragmentActivity.onDestroy(FragmentActivity.java:316)
       at android.support.v7.app.AppCompatActivity.onDestroy(AppCompatActivity.java:158)
       at xxxx.ui.BaseActivity.onDestroy(BaseActivity.java:224)
       at android.app.Activity.performDestroy(Activity.java:6519)
       at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1154)
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4177)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4217)
       at android.app.ActivityThread.access$1500(ActivityThread.java:177)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1502)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:5942)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
       at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:117)

```

--- a/library/src/main/java/com/bumptech/glide/manager/RequestManagerFragment.java
+++ b/library/src/main/java/com/bumptech/glide/manager/RequestManagerFragment.java
@@ -165,7 +165,7 @@ public class RequestManagerFragment extends Fragment {
     super.onAttach(activity);
     try {
       registerFragmentWithRoot(activity);
-    } catch (IllegalArgumentException e) {
+    } catch (IllegalStateException e) {
       // OnAttach can be called after the activity is destroyed, see #497.
       if (Log.isLoggable(TAG, Log.WARN)) {
         Log.w(TAG, "Unable to register fragment with root", e);

--- a/library/src/main/java/com/bumptech/glide/manager/SupportRequestManagerFragment.java
+++ b/library/src/main/java/com/bumptech/glide/manager/SupportRequestManagerFragment.java
@@ -154,7 +154,7 @@ public class SupportRequestManagerFragment extends Fragment {
     super.onAttach(activity);
     try {
       registerFragmentWithRoot(getActivity());
-    } catch (IllegalArgumentException e) {
+    } catch (IllegalStateException e) {
       // OnAttach can be called after the activity is destroyed, see #497.
       if (Log.isLoggable(TAG, Log.WARN)) {
         Log.w(TAG, "Unable to register fragment with root", e);

